packages/edit-site/src/store/actions.js

```diff
@@ -257,22 +257,48 @@ export function setEditedPostContext( context ) {
 export const setPage =
 	( page ) =>
 	async ( { dispatch, registry } ) => {
-		if ( ! page.path && page.context?.postId ) {
-			const entity = await registry
+		let template;
```

Comment:
This is probably the most important changes that should be tested for any regressions or wrong handling in any use case.